### PR TITLE
Remove all things dark-mode

### PR DESCRIPTION
**Pull request by aristath**
## Summary
Removes dark-mode from the theme. A plugin will be created instead, so that we can experiment on iterations easier without blocking the overall theme development. 
Depending on the progress and time, we may re-integrate the plugin inside the theme.

Note: The new repo for dark-mode is on https://github.com/WordPress/twentytwentyone-dark-mode

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

### CODE REVIEW

File: assets/js/customize-preview.js [hunk 1]
```diff
@@ -43,15 +43,6 @@
 				textColor = ! isDark ? 'var(--global--color-dark-gray)' : 'var(--global--color-light-gray)',
 				tableColor = ! isDark ? 'var(--global--color-light-gray)' : 'var(--global--color-dark-gray)';
 
-			// Modify the body class depending on whether this is a dark background or not.
```
Comment (carolinan) [inline assets/js/customize-preview.js]: This class is used for other color settings, like changing the background of the input fields, so it needs to be kept.

File: assets/sass/07-utilities/ie.scss [hunk 2]
```diff
@@ -1,45 +0,0 @@
-.is-IE {
```
Comment (carolinan) [inline assets/sass/07-utilities/ie.scss]: Why is this removed? I thought this was for any dark background, not specifically for the device dark mode

File: classes/class-twenty-twenty-one-custom-colors.php [hunk 3]
```diff
@@ -160,26 +151,4 @@ public static function get_relative_luminance_from_hex( $hex ) {
 		$lum = ( 0.2126 * $red ) + ( 0.7152 * $green ) + ( 0.0722 * $blue );
 		return (int) round( $lum );
 	}
-
```
Comment (carolinan) [inline classes/class-twenty-twenty-one-custom-colors.php]: This body class is used to soften the color of the input field background so needs to be kept.

Comment (aristath) [reply classes/class-twenty-twenty-one-custom-colors.php]: :thinking: where? Ah, I assume it's on another PR that hasn't been merged yet? 'Cause there's nothing like that on `trunk`

Comment (aristath) [reply classes/class-twenty-twenty-one-custom-colors.php]: Added back :+1:

File: assets/js/customize-preview.js (hunk 1 above)
Comment (aristath) [reply assets/js/customize-preview.js]: Added back

File: assets/sass/07-utilities/ie.scss (hunk 2 above)
Comment (aristath) [reply assets/sass/07-utilities/ie.scss]: You're right. added back :+1:
